blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpAggregatedMultiIngestionHandler.java

```diff
@@ -64,107 +66,115 @@ public void handle(ChannelHandlerContext ctx, FullHttpRequest request) {
+                // process valid metrics in collection
+                ListenableFuture<List<Boolean>> futures = processor.apply(collection);
+                List<Boolean> persisteds = futures.get(timeout.getValue(), timeout.getUnit());
+                for (Boolean persisted : persisteds) {
+                    if (!persisted) {
+                        DefaultHandler.sendErrorResponse(ctx, request, "Internal error persisting data", HttpResponseStatus.INTERNAL_SERVER_ERROR);
                         return;
                     }
+                }
 
-                } else {
-                    // no aggregated metric bundles in body, response OK
+                // return OK or MULTI_STATUS response depending if there were validation errors
+                if (errors.isEmpty()) {
+                    // no validation error, response OK
                     DefaultHandler.sendResponse(ctx, request, null, HttpResponseStatus.OK);
                     return;
+                } else {
+                    // has some validation errors, response MULTI_STATUS
+                    DefaultHandler.sendResponse(ctx, request, null, HttpResponseStatus.MULTI_STATUS);
+                    return;
                 }
-            } catch (JsonParseException ex) {
-                log.debug(String.format("BAD JSON: %s", body));
-                log.error(ex.getMessage(), ex);
-                DefaultHandler.sendResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
-            } catch (TimeoutException ex) {
-                DefaultHandler.sendResponse(ctx, request, "Timed out persisting metrics", HttpResponseStatus.ACCEPTED);
-            } catch (Exception ex) {
-                log.debug(String.format("BAD JSON: %s", body));
-                log.error("Other exception while trying to parse content", ex);
-                DefaultHandler.sendResponse(ctx, request, "Failed parsing content", HttpResponseStatus.INTERNAL_SERVER_ERROR);
-            } finally {
-                timerContext.stop();
+
+            } else {
+                // no aggregated metric bundles in body, response OK
+                DefaultHandler.sendResponse(ctx, request, null, HttpResponseStatus.OK);
+                return;
             }
+        } catch (JsonParseException ex) {
+            log.debug(String.format("BAD JSON: %s", body));
+            log.error(ex.getMessage(), ex);
+            DefaultHandler.sendErrorResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
+        } catch (InvalidDataException ex) {
+            log.debug(String.format("Invalid request body: %s", body));
+            log.error(ex.getMessage(), ex);
+            DefaultHandler.sendErrorResponse(ctx, request, ex.getMessage(), HttpResponseStatus.BAD_REQUEST);
+        } catch (TimeoutException ex) {
+            DefaultHandler.sendErrorResponse(ctx, request, "Timed out persisting metrics", HttpResponseStatus.ACCEPTED);
```

Comment:
would we want to log an info or warn for this condition?

So basically we are optimistic here, right?  We are saying that the metrics haven't saved yet, but they will.